centreon/src/Core/TimePeriod/Infrastructure/Repository/DbWriteTimePeriodRepository.php

```diff
@@ -0,0 +1,299 @@
+                    <<<SQL
+                    INSERT INTO `:db`.timeperiod
+                    (tp_name, tp_alias, tp_sunday, tp_monday, tp_tuesday, tp_wednesday,
+                    tp_thursday, tp_friday, tp_saturday)
+                    VALUES (:name, :alias, :sunday, :monday, :tuesday, :wednesday,
+                            :thursday, :friday, :saturday)
+                    SQL
+                )
+            );
+            $this->bindValueOfTimePeriod($statement, $newTimePeriod);
+            $statement->execute();
+            $newTimePeriodId = (int) $this->db->lastInsertId();
+
+            $this->addTimePeriodTemplates($newTimePeriodId, $newTimePeriod->getTemplates());
+            $this->addExtraTimePeriods($newTimePeriodId, $newTimePeriod->getExtraTimePeriods());
+
+            if (! $alreadyInTransaction) {
+                $this->db->commit();
+            }
+            return $newTimePeriodId;
+        } catch (\Throwable $ex) {
+            if (! $alreadyInTransaction) {
+                $this->db->rollBack();
+            }
+            throw $ex;
+        }
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function delete(int $timePeriodId): void
+    {
+        $this->info('Delete time period');
+        $statement = $this->db->prepare(
+            $this->translateDbName('DELETE FROM `:db`.timeperiod WHERE tp_id = :id')
+        );
+        $statement->bindValue(':id', $timePeriodId, \PDO::PARAM_INT);
+        $statement->execute();
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function update(TimePeriod $timePeriod): void
+    {
+        $this->info('Update the time period');
+        $alreadyInTransaction = $this->db->inTransaction();
+        if (! $alreadyInTransaction) {
+            $this->db->beginTransaction();
+        }
+        try {
+            $statement = $this->db->prepare(
+                $this->translateDbName(
+                    <<<SQL
+                    UPDATE `:db`.timeperiod
+                    SET tp_name = :name,
+                        tp_alias = :alias,
+                        tp_monday = :monday,
+                        tp_tuesday = :tuesday,
+                        tp_wednesday = :wednesday,
+                        tp_thursday = :thursday,
+                        tp_friday = :friday,
+                        tp_saturday = :saturday,
+                        tp_sunday = :sunday
+                    WHERE tp_id = :id
+                    SQL
+                )
+            );
+            $this->bindValueOfTimePeriod($statement, $timePeriod);
+            $statement->bindValue(':id', $timePeriod->getId(), \PDO::PARAM_INT);
+            $statement->execute();
+
+            $this->deleteExtraTimePeriods($timePeriod->getId());
+            $this->deleteTimePeriodTemplates($timePeriod->getId());
+
+            $templateIds = array_map(function (Template $template): int {
+                return $template->getId();
+            }, $timePeriod->getTemplates());
```

Comment:
static function ? + up to you arrow_functions
```suggestion
            $templateIds = array_map(
                static fn (Template $template): int => $template->getId(),
                $timePeriod->getTemplates()
            );
```